src/test/java/com/petclinic/selenium/seleniumtest/visit/CalendarUIVisitSeleniumTest.java

```diff
@@ -0,0 +1,231 @@
+
+        //enter authentication information
+        driver.findElement(By.id("email")).sendKeys("admin");
+        driver.findElement(By.id("pwd")).sendKeys("admin");
+        driver.findElement(By.id("button")).click();
+
+
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.linkText("Owners")));
+
+
+
+
+
+        //navigation to visits page
+        driver.findElement(By.linkText("Owners")).click();
+
+
+        WebElement a1 = driver.findElement(By.xpath("//a[@href='#!/owners']"));
+        a1.click();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.linkText("Betty Davis"))); //Will crash here if not wait
+        a1 = driver.findElement(By.linkText("Betty Davis"));
+        a1.click();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.linkText("Add Visit")));
+        a1 = driver.findElement(By.linkText("Add Visit"));
+        a1.click();
+
+
+
+
+
+        //Selecting the vet : "Helen"
+
+        TimeUnit.SECONDS.sleep(2);//special case
+
+        driver.findElement(By.xpath("//*[@id=\"selectedVet\"]/option[1]")).click();
+        TimeUnit.SECONDS.sleep(2);
+        String selectedVet = driver.findElement(By.xpath("//*[@id=\"selectedVet\"]/option[3]")).getText();
+        System.out.println(selectedVet +" : vetname");
+        assertThat(selectedVet, is("Helen Leary"));
+
+
+        // Necessary scroll
+        JavascriptExecutor js = (JavascriptExecutor) driver;
+
+
+        driver.findElement(By.xpath("//*[@id=\"selectedVet\"]/option[3]")).click();
+        TimeUnit.SECONDS.sleep(1);
+
+        Calendar calendar = Calendar.getInstance();
+
+        WebElement currentMonthCalendar = driver.findElement(By.xpath("//*[@id=\'bg\']/div/div/div/ui-view/visits/div[1]/div/div[1]/div/h1"));
+
+
+        //calendar displays the current month
+        assertThat(currentMonthCalendar.getText().toString().toLowerCase(Locale.ROOT), is(calendar.getDisplayName(Calendar.MONTH, Calendar.LONG, Locale.ENGLISH ).toString().toLowerCase(Locale.ROOT)));
+
+        //testing next month button ">"
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@id=\'bg\']/div/div/div/ui-view/visits/div[1]/div/div[1]/i[2]")));
+        driver.findElement(By.xpath("//*[@id=\'bg\']/div/div/div/ui-view/visits/div[1]/div/div[1]/i[2]")).click();
+        WebElement nextMonthCalendar = driver.findElement(By.xpath("//*[@id=\'bg\']/div/div/div/ui-view/visits/div[1]/div/div[1]/div/h1"));
+        assertThat(nextMonthCalendar.getText().toString().toLowerCase(Locale.ROOT), is("november"));
+
+
+        //testing previous month button "<"
+        driver.findElement(By.xpath("//*[@id=\'bg\']/div/div/div/ui-view/visits/div[1]/div/div[1]/i[1]")).click();
+        driver.findElement(By.xpath("//*[@id=\'bg\']/div/div/div/ui-view/visits/div[1]/div/div[1]/i[1]")).click();
+
+        assertThat(nextMonthCalendar.getText().toString().toLowerCase(Locale.ROOT), is("september"));
```

Comment:
Will this test still work if we're not in October? If not, you should make your test more durable.